docs/reference/API.md

```diff
@@ -0,0 +1,464 @@
+
+```jsx
+{ "remote" : Remote_Name,
+  "remote_branch" : Remote_Branch,
+  <"push_prefixes" : Boolean> }
+```
+
+This endpoint pushes deltas from the branch specified in the path to
+the remote repository with the specified remote from the JSON object.
+
+If `"push_prefixes"` is true, then it will also push the prefixes
+associated with the database to the remote.
+
+## Pull
+
+```
+POST http://localhost:6363/api/push/<organization>/<dbid>[/<repo>/branch/<branchid>/]
+```
+JSON API document is:
+
+```jsx
+{ "remote" : Remote_Name,
+  "remote_branch" : Remote_Branch_Name
+}
+```
+
+Fetch layers from `remote`, then attempt a rebase from the remote branch `remote_branch` onto the local branch specified in the URL.
+
+## Branch
+
+```
+POST http://localhost:6363/api/branch/<organization>/<dbid>/<repo>/<new_branchid>
+```
+
+JSON API document is:
+
+```jsx
+{ <"origin" : Remote_Name >
+}
+```
+
+Creates a new branch as specified by the URI, starting from the branch given by `origin` or empty if it is unspecified.
+
+## Create Graph
+
+```
+POST http://localhost:6363/api/graph/<organization>/<dbid>/<repo>/branch/<branchid>/<instance|schema|inference>/<graphid>
+```
+
+This takes a post parameter:
+
+```jsx
+{"commit_info" : { "author" : Author, "message" : Message }}
+```
+
+This API call creates a new graph as specified by the absolute graph descriptor in the URI.
+
+## Delete Graph
+
+```
+DELETE http://localhost:6363/api/graph/<organization>/<dbid>/<repo>/branch/<branchid>/<instance|schema|inference>/<graphid>
+```
+
+This takes a post parameter:
+
+```jsx
+{"commit_info" : { "author" : Author, "message" : Message }}
+```
+This API deletes the graph specified by the absolute graph descriptor in the URI.
+
+
+## Reset
+
+```
+POST http://localhost:6363/api/reset/<organization>/<dbid>/
+POST http://localhost:6363/api/reset/<organization>/<dbid>/local/branch/<branchid>
+```
+
+This takes a post parameter:
+
+```jsx
+{ "commit_descriptor" : Ref }
+```
+
+This API endpoint allows you to set a branch to an arbitrary
+commit. If the branch is left unspecified, it defaults to `"local/main"`.
+
+## Squash
+
+```
+POST http://localhost:6363/api/squash/<organization>/<dbid>/
+POST http://localhost:6363/api/squash/<organization>/<dbid>/local/branch/<branchid>
+```
+
+This takes a post parameter:
+
+```jsx
+{ "commit_info" : Commit_String }
+```
+
+This API endpoint allows you to squashes a branch to a single
+commit. If the branch is left unspecified, it defaults to
+`"local/main"`.
+
+It returns a json object of the form
+
+```jsx
+{ "@type":"api:SquashResponse",
+  "api:commit": New_Commit_Path,
+  "api:old_commit" : Old_Commit_Path,
+  "api:status":"api:success"}
+```
+
+This commit path can be used with reset, to add the commit to a
+branch.
+
+
+## Optimize
+
+```
+POST http://localhost:6363/api/optimize/_system
+POST http://localhost:6363/api/optimize/<organization>/<dbid>
+```
+
+This API endpoint will attempt to optimize the database.
```

Comment:
should mention that this is not recursive. Should mention the different sorts of graphs that can be optimized and what to expect (squash or rollup).